solr/core/src/java/org/apache/solr/cloud/api/collections/Assign.java

```diff
@@ -663,17 +553,14 @@ public AssignStrategyFactory(SolrCloudManager solrCloudManager) {
     public AssignStrategy create(ClusterState clusterState, DocCollection collection) throws IOException, InterruptedException {
       @SuppressWarnings({"unchecked", "rawtypes"})
       List<Map> ruleMaps = (List<Map>) collection.get("rule");
-      String policyName = collection.getStr(POLICY);
       @SuppressWarnings({"rawtypes"})
       List snitches = (List) collection.get(SNITCH);
 
       Strategy strategy = null;
-      if ((ruleMaps == null || ruleMaps.isEmpty()) && !usePolicyFramework(collection, solrCloudManager)) {
+      if ((ruleMaps == null || ruleMaps.isEmpty())) {
         strategy = Strategy.LEGACY;
       } else if (ruleMaps != null && !ruleMaps.isEmpty()) {
```

Comment:
This if clause if the opposite of the previous one, so basically it's just an else. I'd keep this one as the first tested and let the else pick LEGACY.

Reply:
Fixed, thanks.